src/wormhole_transit_relay/transit_server.py

```diff
@@ -283,14 +286,14 @@ def connection_got_token(self, token, new_side, new_tc):
 
                 # drop and stop tracking the rest
                 potentials.remove(old)
-                for (_, leftover_tc) in potentials:
+                for (_, leftover_tc) in potentials.copy():
```

Comment:
The copy here is necessary because calling `disconnect_redundant` will ultimately trigger `connectionLost` method in the `leftover_tc` method, which in turn calls `transitFinished` in the factory.

When using TCP the `dataReceived` method returns before the `connectionLost` method is called in `leftover_tc`, but during testing the string transport will disconnect _immediately_, so `connectionLost` will run _before_ `dataReceived` returns, and an exception will be encountered:

```
RuntimeError: Set changed size during iteration
```

This occurs because the `transitFinished` method _also_ manipulates this set, and during testing this happens as soon as `disconnect_redundant` is called.